Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2700,6 +2703,42 @@ namespace Babylon
                 std::vector<xr::System::Session::Frame::InputSource::Identifier> squeezeStarts{};
                 std::vector<xr::System::Session::Frame::InputSource::Identifier> squeezeEnds{};
 
+                if (frame.EyeTrackerSpace.first)
+                {
+                    if (!m_eyeTrackedSource.first)
+                    {
+                        m_eyeTrackedSource.first = true;
+                        m_eyeTrackedSource.second = Napi::Persistent(Napi::Object::New(env));
+                        m_eyeTrackedSource.second.Set("gazeSpace", Napi::External<xr::System::Session::Frame::Space>::New(env, &frame.EyeTrackerSpace.second));
+
+                        for (const auto& [name, callback] : m_eventNamesAndCallbacks)
+                        {
+                            if (name == JS_EVENT_NAME_EYE_TRACKING_START)
+                            {
+                                Napi::Object obj = m_eyeTrackedSource.second.Value();
+                                callback.Call({obj});
+                            }
+                        }
```

Comment:
this function is already huge, can you move this event firing logic to new functions?